whois-commons/src/main/java/net/ripe/db/whois/common/support/TelnetWhoisClient.java

```diff
@@ -25,52 +30,56 @@ public class TelnetWhoisClient {
     public static final String DEFAULT_HOST = "localhost";
     public static final Charset DEFAULT_CHARSET = StandardCharsets.ISO_8859_1;
     private static final int DEFAULT_TIMEOUT = (int)TimeUnit.MINUTES.toMillis(5);
-
     private final String host;
     private final int port;
 
-    public static String queryLocalhost(final int port, final String query) {
-        TelnetWhoisClient client = new TelnetWhoisClient("127.0.0.1", port);
-        return client.sendQuery(query);
-    }
+    private final Charset charset;
 
-    public static String queryLocalhost(final int port, final String query, final int timeoutMs) {
-        return new TelnetWhoisClient("127.0.0.1", port).sendQuery(query, DEFAULT_CHARSET, timeoutMs);
-    }
 
     public TelnetWhoisClient(final String host) {
-        this.port = DEFAULT_PORT;
-        this.host = host;
+        this(host, DEFAULT_PORT);
     }
 
     public TelnetWhoisClient(final int port) {
-        this.port = port;
-        this.host = DEFAULT_HOST;
+        this(DEFAULT_HOST, port);
     }
 
     public TelnetWhoisClient(final String host, final int port) {
+        this(host, port, DEFAULT_CHARSET);
+    }
+
+    public TelnetWhoisClient(final int port, final Charset charset) {
+        this(DEFAULT_HOST, port, charset);
+    }
+
+    public TelnetWhoisClient(final String host, final int port, final Charset charset) {
         this.port = port;
         this.host = host;
+        this.charset = charset;
     }
 
-    public String sendQuery(final String query) {
-        return sendQuery(query, DEFAULT_CHARSET);
+    public static String queryLocalhost(final int port, final String query) {
+        TelnetWhoisClient client = new TelnetWhoisClient("127.0.0.1", port, DEFAULT_CHARSET);
```

Comment:
Use `(host, port)` constructor, don't specify `DEFAULT_CHARSET` as it repeats the default already in the constructor

Reply:
If I do this I need to turn the method into NON static
then I need to change all the places that are calling this...and there are a lot. This PR will become a monster.
Maybe we can refactor TelnetWhoisClient in a separate PR

Reply:
moved to [DB-5375](https://jira.ripe.net/browse/DB-5375)